internal/rpc/rpc.go

```diff
@@ -240,7 +240,6 @@ func RetryStreamingServerStream[Req, Resp any](
 				} else {
 					// Stream terminated; check if this was caused by an error
 					err = stream.Err()
```

Comment:
Should we check if the error type is actually a `connect: connection refused` error? I'm guessing there could be other error types here where we _do_ want `log.Warn`